src/ruby/puppet-server-lib/puppet/server/oids.rb

```diff
@@ -0,0 +1,31 @@
+module Puppet::Server::Oids
+  def self.load_custom_oid_file(custom_oid_file, map_key='oid_mapping')
+    if File.exists?(custom_oid_file) && File.readable?(custom_oid_file)
```

Comment:
ok that's fair. i don't know how we've done this in the past, but keeping it as close to the source as possible sounds reasonable.